src/EventStore.Core/Messaging/ContinuationEnvelope.cs

```diff
@@ -36,4 +36,22 @@ public class ContinuationEnvelope : IEnvelope {
 			catch (OperationCanceledException) {}
 		}
 	}
+
+	public class AsyncTaskEnvelope : IEnvelope {
```

Comment:
probably in its own file

and probably worth a comment at the top of this that it needs to be used carefully because the callback is called initially on the thread that is replying to the message so must be quick

Reply:
fixed, comment added :+1: